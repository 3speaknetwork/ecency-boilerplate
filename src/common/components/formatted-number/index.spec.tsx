import React from "react";

import FormattedNumber from "./index";
import renderer from "react-test-renderer";

it("(1) FormattedNumber", () => {
  const props = {
    value: 1250,
  };

  const component = renderer.create(<FormattedNumber {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it("(2) FormattedNumber - with fraction digits", () => {
  const props = {
    value: 1250,
    fractionDigits: 2,
  };

  const component = renderer.create(<FormattedNumber {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it("(3) FormattedNumber - with prefix", () => {
  const props = {
    value: 100,
    prefix: "$",
  };

  const component = renderer.create(<FormattedNumber {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it("(4) FormattedNumber - with suffix", () => {
  const props = {
    value: 100,
    prefix: "HIVE",
  };

  const component = renderer.create(<FormattedNumber {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
