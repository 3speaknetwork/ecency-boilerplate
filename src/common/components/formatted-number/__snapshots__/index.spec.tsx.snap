// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) FormattedNumber 1`] = `"1,250.000"`;

exports[`(2) FormattedNumber - with fraction digits 1`] = `"1,250.00"`;

exports[`(3) FormattedNumber - with prefix 1`] = `"$ 100.000"`;

exports[`(4) FormattedNumber - with suffix 1`] = `"HIVE 100.000"`;
